pkg/querier/timeseries_series_set_test.go

```diff
@@ -78,49 +84,82 @@ func TestTimeSeriesIterator(t *testing.T) {
 					TimestampMs: 1234,
 				},
 				{
-					Value:       3.14,
+					Value:       3.15,
 					TimestampMs: 1235,
 				},
 				{
-					Value:       3.14,
+					Value:       3.16,
 					TimestampMs: 1236,
 				},
+				{
+					Value:       3.17,
+					TimestampMs: 1237,
+				},
+			},
+			Histograms: []mimirpb.Histogram{
+				mimirpb.FromHistogramToHistogramProto(1232, generateTestHistogram(7)),
+				mimirpb.FromFloatHistogramToHistogramProto(1233, generateTestFloatHistogram(8)),
 			},
 		},
 	}
 
 	it := ts.Iterator(nil)
```

Comment:
I feel we could use a slice of expected stuff and use a loop to do the matching to avoid a lot of duplication below. In case of seek, we could make it check against a partial slice. But if you feel it is easier this way, then I have no problem :)

Reply:
I don't want to reinvent this testcase right now, plus I've added a case where we test two things after Next call, so I'd rather defer this question.

Reply:
rewrote using test util functions to be more concise